app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardClustersStep/MappingWizardClustersStepReducer.js

```diff
@@ -10,9 +14,35 @@ const initialState = Immutable({
   targetClusters: [],
   isFetchingTargetClusters: false,
   isRejectedTargetClusters: false,
-  errorTargetClusters: null
+  errorTargetClusters: null,
+  hostsByClusterID: {},
+  isFetchingHostsQuery: false,
+  isRejectedHostsQuery: false,
+  errorHostsQuery: null
 });
 
+const getHostsByClusterID = hostsQueryFulfilledAction => {
+  const {
+    meta: { hostIDsByClusterID },
+    payload: { data }
+  } = hostsQueryFulfilledAction;
+  const hostsByID = data.results.reduce(
+    (newObject, host) => ({
+      ...newObject,
+      [host.id]: host
+    }),
+    {}
+  );
+  const hostsByClusterID = Object.keys(hostIDsByClusterID).reduce(
+    (newObject, clusterID) => ({
+      ...newObject,
+      [clusterID]: hostIDsByClusterID[clusterID].map(hostID => hostsByID[hostID])
+    }),
+    {}
+  );
+  return hostsByClusterID;
+};
```

Comment:
It might be nice to move these helpers to a different file to reduce clutter, but not a big deal.  I think we do this in some of the other reducers